compiler/plugins/target/ROCM/builtins/ukernel/iree_uk_amdgpu_multi_mma_mfma_i32_16x16x32_i8.c

```diff
@@ -61,6 +61,13 @@ int32_t iree_uk_amdgpu_multi_mma_mfma_i32_16x16x32_i8_query_shared_memory_bytes(
 // into shared memory), care is taken to subtract the thread-relative offset,
 // which is computed from the thread id.
 //
+// The C matrix may come from either global, shared, or thread-local memory. If
+// the C matrix comes from global or shared memory, then the offsets into the
+// C matrix must be strided because the C inner tile is shared accross the
+// threads within a workgroup. Otherwise, the C matrix buffer is a contiguous
+// block of memory, and offsets are not strided. The stride between the output
+// tiles of each unrolled intrinsic is passed to the ukernel for this reason.
+//
```

Comment:
Can we find a wording that is more directly matching what actually matters (the layouts) and leaves out aspects that are not directly relevant (types of memory)?